Spigot-Server-Patches/0329-Add-hand-to-bucket-events.patch

```diff
@@ -0,0 +1,111 @@
+ 
+             if (event.isCancelled()) {
+                 return false;
+diff --git a/src/main/java/net/minecraft/server/ItemBucket.java b/src/main/java/net/minecraft/server/ItemBucket.java
+index 4355ef52c..6834b7fba 100644
+--- a/src/main/java/net/minecraft/server/ItemBucket.java
++++ b/src/main/java/net/minecraft/server/ItemBucket.java
+@@ -37,7 +37,7 @@ public class ItemBucket extends Item {
+                     if (iblockdata.getBlock() instanceof IFluidSource) {
+                         // CraftBukkit start
+                         FluidType dummyFluid = ((IFluidSource) iblockdata.getBlock()).a(DummyGeneratorAccess.INSTANCE, blockposition, iblockdata);
+-                        PlayerBucketFillEvent event = CraftEventFactory.callPlayerBucketFillEvent(entityhuman, blockposition.getX(), blockposition.getY(), blockposition.getZ(), null, itemstack, dummyFluid.b());
++                        PlayerBucketFillEvent event = CraftEventFactory.callPlayerBucketFillEvent(entityhuman, blockposition.getX(), blockposition.getY(), blockposition.getZ(), null, itemstack, dummyFluid.b(), enumhand); // Paper - add enumHand
+ 
+                         if (event.isCancelled()) {
+                             return new InteractionResultWrapper(EnumInteractionResult.FAIL, itemstack);
+@@ -63,7 +63,7 @@ public class ItemBucket extends Item {
+                     iblockdata = world.getType(blockposition);
+                     BlockPosition blockposition1 = this.a(iblockdata, blockposition, movingobjectposition);
+ 
+-                    if (this.a(entityhuman, world, blockposition1, movingobjectposition, movingobjectposition.direction, blockposition, itemstack)) { // CraftBukkit
++                    if (this.a(entityhuman, world, blockposition1, movingobjectposition, movingobjectposition.direction, blockposition, itemstack, enumhand)) { // CraftBukkit // Paper - add enumHand
+                         this.a(world, itemstack, blockposition1);
+                         if (entityhuman instanceof EntityPlayer) {
+                             CriterionTriggers.y.a((EntityPlayer) entityhuman, blockposition1, itemstack);
+@@ -113,10 +113,10 @@ public class ItemBucket extends Item {
+ 
+     // CraftBukkit start
+     public boolean a(@Nullable EntityHuman entityhuman, World world, BlockPosition blockposition, @Nullable MovingObjectPosition movingobjectposition) {
+-        return a(entityhuman, world, blockposition, movingobjectposition, null, null, null);
++        return a(entityhuman, world, blockposition, movingobjectposition, null, null, null, null); // Paper - add enumHand
+     }
+ 
+-    public boolean a(EntityHuman entityhuman, World world, BlockPosition blockposition, @Nullable MovingObjectPosition movingobjectposition, EnumDirection enumdirection, BlockPosition clicked, ItemStack itemstack) {
```

Comment:
sadly we also need to maintain all old methods to avoid breaking nms using plugins :(